### ISSUE TITLE
Spray accuracy setting doesn't seem to work right...

### ISSUE THREAD

**Issue by Twilight13** [LemADEC/GunCus#11]
GunCustomization-1.7.10-3.45.0.22:

When using an addon pack, you can set the "Spray" variable for bullets, controlling their maximum accuracy, but even when set at 100%, the bullets have a considerable inaccuracy to them. This is using the Wholy addon. If I have it set to 100%, I expect it to hit exactly where I fire, but it does not. Strangely enough, in Creative mode, accuracy _is_ pinpoint, 100% accurate, exactly where I aim, as I would expect 100% to be, so it can be set that way, it's just that "Spray=100" doesn't seem to do it.

**Comment by LemADEC** [comment]
Bullet spray accuracy is only a minimum inaccuracy compared with the current player status.
Laser addon, Scoping (sneak), Riffled barrel addon, Sprinting, Bipod addon and recent shooting will change the player accuracy.
Also, when in creative, your accuracy is forced to 100%.
Those equations are 'legacy', so there's probably a bug in there but we'll need more details to fix it.

**Comment by Twilight13** [comment]
I understand having all the different factors affecting the current accuracy, but it would be nice if you could set a "base" accuracy level, and then have the other factors affect things from there. That way you can set accuracy as good or as bad as you like, and then still let the other things like laser sights affect it.

Also, when I did some (Admittedly quick tests) between accuracy with a weapon with laser sights or a bipod and without, the accuracy didn't seem very different. It was better, a bit, but not as much as I was expecting. This is why it would be nice to be able to specify a base accuracy, and/or even the accuracy effects of sprinting, scopes, etc. Still, that might take some work and be outside the current scope of things. I'm not trying to complain...

Incidentally, thanks for all the work you've done on this mod and responding to my issues so quickly. I'm surprised this mod isn't more well known.

**Comment by LemADEC** [comment]
Currently, the bullet.spray value is the maximum accuracy of the bullet while the latest comes from the player accuracy which is changing depending on history (recent shoots), current player state (jumping, moving, scoping, walking, etc.) and the gun upgrades.
Because it's capped instead of being factored, there's some unexpected effects including:
1- if you have a low accuracy bullet, sprint/jumping won't change much to you.
2- gun upgrades have more effect as you shoot more bullets. In other words, the first bullet accuracy isn't influenced by your upgrades.
To fix the equation, we need to redefine all the factors values which is the hard part.

Furthermore, the accuracy effect on the bullet is approximated in rectangular space, so it has influence on the bullet speed, which gives strange effects on slow bullets like the flame thrower ones.

**Comment by LemADEC** [comment]
We did a pass on accuracy computation by:
- boosting barrels and attachment stabilisation bonus,
- adding stability bonus to 5 more barrels and attachments,
- adding minimum destabilisation for very low damage bullets (BB minigun anyone?),
- fixing stability cap application.
  This gives more use to gun addons other than cosmetics and adds more stability over time.

We've also added a parameter to adjust player accuracy impact. Setting it above 100 with spray at 100 will make a perfect hit bullet.
